src/libAtomVM/interop.h

```diff
@@ -108,6 +111,61 @@ static inline term interop_kv_get_value(term kv, AtomString key, GlobalContext *
     return interop_kv_get_value_default(kv, key, term_invalid_term(), glb);
 }
 
+//
+// Use this macro to generate code to fold over elements (integer values or binaries)
+// in an `iodata()` structure.
+//
+#define INTEROP_IODATA_FOLD(t, fold_fun, accum)                                                     \
```

Comment:
According to `elf_diff`, the macro version when used for `interop_iolist_size`, on x86_64 produces nearly the same code or maybe a 11 bytes longer version compared the 2 inlines version that I tested.
I general macros are even more complex to maintain, so unless it produces a way optimal function for interop_write_iolist, it doesn't look like a real improvement.